game/world/worldobjects.cpp

```diff
@@ -408,10 +408,15 @@ void WorldObjects::triggerEvent(const TriggerEvent &e) {
   }
 
 bool WorldObjects::execTriggerEvent(const TriggerEvent& e) {
+  if(e.trigger!=nullptr) {
+    e.trigger->processEvent(e);
+    return true;
+    }
   bool emitted=false;
   for(auto& i:triggers) {
     auto& t = *i;
     if(t.name()==e.target) { // NOTE: trigger name is not unique - more then one trigger can be activated
+      e.trigger = i;
```

Comment:
This doesn't seem to be correct: when `t.name()` matches multiple receivers, all of them should receive event.
After this change they will but only one of them will populate `e.trigger`.